### DISCUSSION TITLE
type error with useMenu example from docs

### DISCUSSION THREAD

**Discussion by bstro** [adobe/react-spectrum#3440] [Q&A] [answered]
I'm running into a type error when trying to type the  [useMenu example](https://react-spectrum.adobe.com/react-aria/useMenu.html#example) from the docs— I'm seeing a `Type 'unknown' is not assignable to type 'object'` error when trying to render the `<Menu>` component

https://codesandbox.io/s/frosty-hofstadter-n7rf4w?file=/src/components/MenuButton.tsx:952-1005

It looks like there's also a ts error on https://codesandbox.io/s/frosty-hofstadter-n7rf4w?file=/src/components/Menu.tsx:665-708 — i'm not experiencing this locally, but perhaps it is related.

**Comment by snowystinger** [comment]
Might be ts strict? Could you check what the ts config is?

**Reply by bstro** [reply]
Yep, I enable strict for all my ts projects. Is it recommended to disable strict to use React-ARIA?
edit: disabling strict mode does not resolve the TS error, however

**Reply by snowystinger** [reply]
It's not recommended, no, though we are not currently strict, so sometimes there are issues which is why I suspected it. We are working on becoming strict, we just aren't there yet.

Thank you for checking on it. Unfortunately I'm out sick today, so I'll have to look more at this later. @LFDanLu if you have some bandwidth.

**Reply by bstro** [reply]
Thanks @snowystinger! Hope ya feel better soon!

**Reply by LFDanLu** [reply]
I believe you have to supply `<object>` to useMenuTrigger: https://codesandbox.io/s/restless-paper-qylnk7?file=/src/components/MenuButton.tsx:618-684. That way you match with your type for TreeProps in Menu. As for the spread operation for collection, not sure why that is complaining, you should be able to spread an iterator. Perhaps a ES target issue in the sandbox.

**Reply by bstro** [reply]
Thank you. This worked great. I'm not experiencing that iterator error locally, so I'm good to go. Many thanks to you and @snowystinger

**Comment by bstro** [comment, marked as answer]
quoting [@LFDanLu](https://github.com/adobe/react-spectrum/discussions/3440#discussioncomment-3470014) so I can mark as answer:
> I believe you have to supply <object> to useMenuTrigger: https://codesandbox.io/s/restless-paper-qylnk7?file=/src/components/MenuButton.tsx:618-684. That way you match with your type for TreeProps in Menu. As for the spread operation for collection, not sure why that is complaining, you should be able to spread an iterator. Perhaps a ES target issue in the sandbox.
